azurerm_function_app_flex_consumption: Creation of two apps concurrently using the same Subnet fails

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->

We are using the resource `azurerm_function_app_flex_consumption` and configure a `virtual_network_subnet_id`.
We have multiple function apps which use the same subnet.  

If two apps, which are using the same subnet, are getting created at the same time, we see this error:

```
│ Error: creating App Service (Subscription: "<subscriptionId>"
│ Resource Group Name: "ps-functionapp"
│ Site Name: "func-functionadsfrettzjjgg1"): performing CreateOrUpdate: unexpected status 409 (409 Conflict) with response: {"Code":"Conflict","Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'","Target":null,"Details":[{"Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'"},{"Code":"Conflict"},{"ErrorEntity":{"ExtendedCode":"70006","MessageTemplate":"The ServiceAssociationLink (SAL) related operation failed with error '{0}'","Parameters":["Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]"],"Code":"Conflict","Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'"}}],"Innererror":null}
```

If we run the `terraform apply` again, creation succeeds.

I guess it is a race condition while the service association link is created.


### Terraform Version

1.46.6

### AzureRM Provider Version

4.66.0

### Affected Resource(s)/Data Source(s)

azurerm_function_app_flex_consumption

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "4.66.0"
    }
  }
  backend "local" {}
}

provider "azurerm" {
  subscription_id = "tbd"
  features {
  }
  storage_use_azuread = true
}


resource "azurerm_resource_group" "example" {
  name     = "ps-functionapp"
  location = "GermanyWestCentral"
  lifecycle {
    ignore_changes = [tags]
  }
}

resource "azurerm_virtual_network" "example" {
  name                = "ps-example-network"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  address_space       = ["10.0.0.0/16"]

  subnet {
    name             = "subnet1"
    address_prefixes = ["10.0.1.0/24"]
    delegation=[{
      name = "Microsoft.App/environments"

      service_delegation =[{
        name = "Microsoft.App/environments"
        actions = [
          "Microsoft.Network/virtualNetworks/subnets/join/action"
        ]
      }]
    }]
  }

  lifecycle {
    ignore_changes = [tags]
  }
}

locals {
  # will fail when both are created at the same time, but succeeds after retry or if applied one at a time
  functions= toset(["functionadsfrettzjjgg1", "functionadsfrettzjjgg3"])
}

resource "azurerm_service_plan" "example" {
  for_each            =local.functions
  name                = "asp-${each.key}"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  os_type             = "Linux"
  sku_name            = "FC1" # Flex Consumption plan

  lifecycle {
    ignore_changes = [tags]
  }
}

resource "azurerm_storage_account" "example" {
  name                     = "sapssomerandasdfsdf"
  location                 = azurerm_resource_group.example.location
  resource_group_name      = azurerm_resource_group.example.name
  account_tier             = "Standard"
  account_replication_type = "LRS"

  https_traffic_only_enabled        = true
  default_to_oauth_authentication   = true
  public_network_access_enabled     = true
  shared_access_key_enabled         = false
  allow_nested_items_to_be_public   = false
  infrastructure_encryption_enabled = true

  lifecycle {
    ignore_changes = [tags]
  }
}

resource "azurerm_function_app_flex_consumption" "example" {
  for_each            =local.functions
  name                = "func-${each.key}"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  service_plan_id     = azurerm_service_plan.example[each.key].id
  https_only          = true

  runtime_name    = "python"
  runtime_version = "3.11"

  # VNet integration for outgoing network traffic
  virtual_network_subnet_id ="${azurerm_virtual_network.example.id}/subnets/subnet1"

  public_network_access_enabled = false

  identity {
    type = "SystemAssigned"
  }

  site_config {}

  lifecycle {
    ignore_changes = [tags]
  }
  storage_authentication_type = "SystemAssignedIdentity"
  storage_container_endpoint  = azurerm_storage_account.example.primary_blob_host
  storage_container_type      = "blobContainer"
}
```

### Debug Output/Panic Output

```shell
2026-04-01T17:30:29.063+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: [DEBUG] PUT https://management.azure.com/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/providers/Microsoft.Web/sites/func-functionadsfrettzjjgg1?api-version=2023-12-01
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/providers/Microsoft.Web/sites/func-functionadsfrettzjjgg1?api-version=2023-12-01: 
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: HTTP/2.0 409 Conflict
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: Content-Length: 1781
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: Cache-Control: no-cache
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: Content-Type: application/json; charset=utf-8
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: Date: Wed, 01 Apr 2026 15:30:32 GMT
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: Expires: -1
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: Pragma: no-cache
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: Strict-Transport-Security: max-age=31536000; includeSubDomains
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: X-Aspnet-Version: 4.0.30319
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: X-Cache: CONFIG_NOCACHE
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: X-Content-Type-Options: nosniff
2026-04-01T17:30:32.077+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: X-Ms-Correlation-Request-Id: cd298399-df7f-928d-500b-ff52924e5212
2026-04-01T17:30:32.078+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: X-Ms-Operation-Identifier: tenantId=<tenantId>,objectId=e2ee177a-43b7-404e-8ffb-9d0f7f96990d/germanywestcentral/456fd549-f2d8-424b-aaaa-bca42bc99d24
2026-04-01T17:30:32.078+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: X-Ms-Ratelimit-Remaining-Subscription-Resource-Requests: 799
2026-04-01T17:30:32.078+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: X-Ms-Request-Id: bfccbcfe-2b9a-4d40-ad76-65f10bd6d379
2026-04-01T17:30:32.078+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: X-Ms-Routing-Request-Id: GERMANYWESTCENTRAL:20260401T153032Z:bfccbcfe-2b9a-4d40-ad76-65f10bd6d379
2026-04-01T17:30:32.078+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: X-Msedge-Ref: Ref A: 0E6AF29F8C564939A2BC297D72919947 Ref B: FRA261110504025 Ref C: 2026-04-01T15:30:29Z
2026-04-01T17:30:32.078+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: X-Powered-By: ASP.NET
2026-04-01T17:30:32.078+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5
2026-04-01T17:30:32.078+0200 [DEBUG] provider.terraform-provider-azurerm_v4.66.0_x5: {"Code":"Conflict","Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'","Target":null,"Details":[{"Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'"},{"Code":"Conflict"},{"ErrorEntity":{"ExtendedCode":"70006","MessageTemplate":"The ServiceAssociationLink (SAL) related operation failed with error '{0}'","Parameters":["Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]"],"Code":"Conflict","Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'"}}],"Innererror":null}
2026-04-01T17:30:32.078+0200 [TRACE] provider.terraform-provider-azurerm_v4.66.0_x5: Called downstream: @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/resource.go:981 tf_mux_provider="*schema.GRPCProviderServer" tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_resource_type=azurerm_function_app_flex_consumption @module=sdk.helper_schema tf_req_id=6ace6b32-d9e3-3961-7300-1d7a4147236a tf_rpc=ApplyResourceChange timestamp="2026-04-01T17:30:32.077+0200"
2026-04-01T17:30:32.078+0200 [TRACE] provider.terraform-provider-azurerm_v4.66.0_x5: Received downstream response: tf_proto_version=5.10 tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_duration_ms=5926 tf_req_id=6ace6b32-d9e3-3961-7300-1d7a4147236a tf_resource_type=azurerm_function_app_flex_consumption tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:61 @module=sdk.proto diagnostic_error_count=1 diagnostic_warning_count=0 timestamp="2026-04-01T17:30:32.078+0200"
2026-04-01T17:30:32.078+0200 [ERROR] provider.terraform-provider-azurerm_v4.66.0_x5: Response contains error diagnostic: tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/diag/diagnostics.go:58
  diagnostic_summary=
  | creating App Service (Subscription: "<subscriptionId>"
  | Resource Group Name: "ps-functionapp"
  | Site Name: "func-functionadsfrettzjjgg1"): performing CreateOrUpdate: unexpected status 409 (409 Conflict) with response: {"Code":"Conflict","Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'","Target":null,"Details":[{"Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'"},{"Code":"Conflict"},{"ErrorEntity":{"ExtendedCode":"70006","MessageTemplate":"The ServiceAssociationLink (SAL) related operation failed with error '{0}'","Parameters":["Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]"],"Code":"Conflict","Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'"}}],"Innererror":null}
   tf_proto_version=5.10 tf_req_id=6ace6b32-d9e3-3961-7300-1d7a4147236a @module=sdk.proto
  diagnostic_detail=
  | creating App Service (Subscription: "<subscriptionId>"
  | Resource Group Name: "ps-functionapp"
  | Site Name: "func-functionadsfrettzjjgg1"): performing CreateOrUpdate: unexpected status 409 (409 Conflict) with response: {"Code":"Conflict","Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'","Target":null,"Details":[{"Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'"},{"Code":"Conflict"},{"ErrorEntity":{"ExtendedCode":"70006","MessageTemplate":"The ServiceAssociationLink (SAL) related operation failed with error '{0}'","Parameters":["Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]"],"Code":"Conflict","Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'"}}],"Innererror":null}
   diagnostic_severity=ERROR tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_resource_type=azurerm_function_app_flex_consumption timestamp="2026-04-01T17:30:32.078+0200"
```

### Expected Behaviour

Both functions apps are getting created.

### Actual Behaviour


```
│ Error: creating App Service (Subscription: "<subscriptionId>"
│ Resource Group Name: "ps-functionapp"
│ Site Name: "func-functionadsfrettzjjgg1"): performing CreateOrUpdate: unexpected status 409 (409 Conflict) with response: {"Code":"Conflict","Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'","Target":null,"Details":[{"Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'"},{"Code":"Conflict"},{"ErrorEntity":{"ExtendedCode":"70006","MessageTemplate":"The ServiceAssociationLink (SAL) related operation failed with error '{0}'","Parameters":["Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]"],"Code":"Conflict","Message":"The ServiceAssociationLink (SAL) related operation failed with error 'Unable to integrate function app [func-functionadsfrettzjjgg1] with subnet [/subscriptions/<subscriptionId>/resourceGroups/ps-functionapp/virtualNetworks/ps-example-network/subnets/subnet1] due to error: [Failed to acquire ServiceAssociationLinkLease. Another thread or instance may be holding it.]'"}}],"Innererror":null}
```

### Steps to Reproduce

See the attached terraform example. Just run `terraform apply` and it should fail

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/appservice/function_app_flex_consumption_resource.go
+++ b/internal/services/appservice/function_app_flex_consumption_resource.go
@@ -841,6 +841,9 @@ func (r FunctionAppFlexConsumptionResource) Update() sdk.ResourceFunc {
 					model.Properties.VirtualNetworkSubnetId = empty
 				} else {
 					model.Properties.VirtualNetworkSubnetId = pointer.To(subnetId)
+
+					locks.ByID(subnetId)
+					defer locks.UnlockByID(subnetId)
 				}
 			}
 

--- a/internal/services/appservice/function_app_flex_consumption_resource.go
+++ b/internal/services/appservice/function_app_flex_consumption_resource.go
@@ -18,6 +18,7 @@ import (
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/web/2023-01-01/resourceproviders"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/web/2023-12-01/webapps"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/helpers"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/validate"
@@ -533,6 +534,8 @@ func (r FunctionAppFlexConsumptionResource) Create() sdk.ResourceFunc {
 
 			if functionAppFlexConsumption.VirtualNetworkSubnetID != "" {
 				siteEnvelope.Properties.VirtualNetworkSubnetId = pointer.To(functionAppFlexConsumption.VirtualNetworkSubnetID)
+				locks.ByID(functionAppFlexConsumption.VirtualNetworkSubnetID)
+				defer locks.UnlockByID(functionAppFlexConsumption.VirtualNetworkSubnetID)
 			}
 
 			if functionAppFlexConsumption.ClientCertExclusionPaths != "" {

--- a/internal/services/appservice/function_app_flex_consumption_resource_test.go
+++ b/internal/services/appservice/function_app_flex_consumption_resource_test.go
@@ -632,7 +632,7 @@ func TestAccFunctionAppFlexConsumption_vNetIntegrationWithVnetProperties(t *test
 
 	data.ResourceTest(t, r, []acceptance.TestStep{
 		{
-			Config: r.vNetIntegration_subnetWithVnetProperties(data),
+			Config: r.vNetIntegration(data),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 				check.That(data.ResourceName).Key("virtual_network_subnet_id").MatchesOtherKey(
@@ -1683,16 +1683,16 @@ resource "azurerm_function_app_flex_consumption" "test" {
 `, r.template(data), data.RandomInteger)
 }
 
-func (r FunctionAppFlexConsumptionResource) vNetIntegration_subnetWithVnetProperties(data acceptance.TestData) string {
+func (r FunctionAppFlexConsumptionResource) vNetIntegration(data acceptance.TestData) string {
 	return fmt.Sprintf(`
 provider "azurerm" {
   features {}
 }
 
-%s
+%[1]s
 
 resource "azurerm_virtual_network" "test" {
-  name                = "acctest-vnet-%d"
+  name                = "acctest-vnet-%[2]d"
   address_space       = ["10.0.0.0/16"]
   location            = azurerm_resource_group.test.location
   resource_group_name = azurerm_resource_group.test.name
@@ -1720,8 +1720,16 @@ resource "azurerm_subnet" "test1" {
   }
 }
 
+resource "azurerm_service_plan" "test2" {
+  name                = "acctestASP2-%[2]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  os_type             = "Linux"
+  sku_name            = "FC1"
+}
+
 resource "azurerm_function_app_flex_consumption" "test" {
-  name                      = "acctest-LFA-%d"
+  name                      = "acctest-LFA-%[2]d"
   location                  = azurerm_resource_group.test.location
   resource_group_name       = azurerm_resource_group.test.name
   service_plan_id           = azurerm_service_plan.test.id
@@ -1740,7 +1748,28 @@ resource "azurerm_function_app_flex_consumption" "test" {
     vnet_route_all_enabled = true
   }
 }
-`, r.template(data), data.RandomInteger, data.RandomInteger)
+
+resource "azurerm_function_app_flex_consumption" "test2" {
+  name                      = "acctest-LFA2-%[2]d"
+  location                  = azurerm_resource_group.test.location
+  resource_group_name       = azurerm_resource_group.test.name
+  service_plan_id           = azurerm_service_plan.test2.id
+  virtual_network_subnet_id = azurerm_subnet.test1.id
+
+  storage_container_type      = "blobContainer"
+  storage_container_endpoint  = "${azurerm_storage_account.test.primary_blob_endpoint}${azurerm_storage_container.test.name}"
+  storage_authentication_type = "StorageAccountConnectionString"
+  storage_access_key          = azurerm_storage_account.test.primary_access_key
+  runtime_name                = "node"
+  runtime_version             = "22"
+  maximum_instance_count      = 100
+  instance_memory_in_mb       = 2048
+
+  site_config {
+    vnet_route_all_enabled = true
+  }
+}
+`, r.template(data), data.RandomInteger)
 }
 
 func (FunctionAppFlexConsumptionResource) template(data acceptance.TestData) string {

